bioconda_utils/bioconda_utils-conda_build_config.yaml

```diff
@@ -29,28 +29,27 @@ bamtools:
 r_base:
   - 4.0.*
 python:
+  - 3.9.*  *_cpython
   - 3.8.*  *_cpython
   - 3.7.*  *_cpython
   - 3.6.*  *_cpython
   - 2.7.*  *_cpython
+# conda-forge only has 3 python versions, so we need to modify the length of the other zip_keys in its group
+python_impl:
+  - cpython
+  - cpython
+  - cpython
+  - cpython
+  - cpython
+numpy:
+  - 1.19
+  - 1.16
+  - 1.16
+  - 1.16
+  - 1.16
```

Comment:
I wonder if we have to do a similar dance as with `python`/`r_base` above regarding https://github.com/conda/conda-build/issues/3974 ..

Reply:
Hm.. I find things around merged config + zip_keys too confusing (code is not without bugs and I don't understand all intentions).
How about we avoid all of that and do things manually? I.e., not use multiple "exclusive config files" for now, but just copy conda-forge's, comment out what we want to override and pre- or append our pinnings in the same file. Not ideal, but probably much safer...

Reply:
What do you think about this one? I'm rather concerned that `conda-build` will manage to collapse these list again :/.
IDK if it will work because we only duplicate an entry but not add a different one here (adding a different one made `conda-build` barf before).

Reply:
Oh, and some hours ago conda-forge updated to `numpy=1.17` -- so, probably best to update the conda-forge-pinnings version again ;).

Reply:
Good to know, we might as well update then. The main issue with copying over their pinnings and modifying them is that it's easier to lose track of what our differences are. If you're really concerned maybe we merge and tag and then manually run a couple packages for testing. Since it's numpy and python we can use pybigwig as a test (it uses both with C so it needs the pinning).

Reply:
I'll install this locally in a new environment and try this there.

Reply:
It's good I started looking into this, we can't go above numpy 1.16 with python 2.7

Reply:
Hmm, this doesn't want to build against python 3.9

Reply:
This is definitely a problem. If we keep only python 2.7 and 3.9 then the zip_keys work properly, 2.7 builds use numpy 1.16 and 3.9 builds get numpy 1.19. However, once we add in python 3.6, 3.7, or 3.8, then only those python builds get used and the others ignored. They use the right numpy versions, but that's not terribly helpful.

Reply:
Ah, bummer. Yep, that sounds like the exact same "`conda-build` takes the intersection of values from `conda_build_config.yaml` and `bioconda-conda_build_config.yaml`"-issue.
So, for `numpy` itself we could go with the "append `.*`" workaround to get distinct sets.
As for `python_impl`: That one is used as a verbatim string (e.g., with `blabla  # [python_impl == "cpython"]` in conda-forge recipes). We don't use `python_impl` in our recipes currently, so it should generally not be important yet. And, more significantly, we only have a single `cpython` value in the configs for now, so the intersection remains just `cpython`.

TL;DR We could use `numpy: [ 1.16.*, 1.17.*, ... ]` for now. I'll add that change.

Reply:
Good call, I'll run that locally to see if it resolves the issue.

Reply:
Yup, looks like that solved it locally